pkg/controller/planexecution/planexecution_controller.go

```diff
@@ -165,26 +165,28 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		&handler.EnqueueRequestsFromMapFunc{
 			ToRequests: handler.ToRequestsFunc(
 				func(a handler.MapObject) []reconcile.Request {
+					// instance in instance -> reconcile plan for outer instance
 					requests := mapToOwningInstanceActivePlan(a)
-					if len(requests) == 0 {
-						inst := &kudov1alpha1.Instance{}
-						err = mgr.GetClient().Get(context.TODO(), client.ObjectKey{
-							Name:      a.Meta.GetName(),
-							Namespace: a.Meta.GetNamespace(),
-						}, inst)
-
-						if err == nil {
-							// for every updated/added instance also trigger reconcile for its active plan
-							requests = append(requests, reconcile.Request{
-								NamespacedName: types.NamespacedName{
-									Name:      inst.Status.ActivePlan.Name,
-									Namespace: inst.Status.ActivePlan.Namespace,
-								},
-							})
-						} else {
-							log.Printf("PlanExecutionController: received event from Instance %s/%s but instance of that name does not exist", a.Meta.GetNamespace(), a.Meta.GetName())
-						}
+
```

Comment:
the diff is really weird, but the only change is that I removed the `if len(requests) == 0` and now I trigger it for inner as well as outer all the time. More the merrier, right! :D . 

But I really saw that in the test logs, that we missed triggering reconciliation for the inner instance and because of that the plan never finished.